src/board.py

```diff
@@ -20,16 +20,18 @@ class Board:
         Item at [7][0] is referenced as ["a1"].
     """
 
-    index_to_file = {index_: file_ for index_, file_ in zip(range(8), "abcdefgh")}
-    file_to_index = {file_: index_ for index_, file_ in zip(range(8), "abcdefgh")}
-    index_to_rank = {index_: rank_ for index_, rank_ in zip(range(8), "87654321")}
-    rank_to_index = {rank_: index_ for index_, rank_ in zip(range(8), "87654321")}
+    index_to_file = {index_: file_ for index_, file_ in zip(range(8), "abcdefgh")}  # translate range index to file in chess
+    file_to_index = {file_: index_ for index_, file_ in zip(range(8), "abcdefgh")}  # translate file in chess to range index
+    index_to_rank = {index_: rank_ for index_, rank_ in zip(range(8), "87654321")}  # translate range index to rank in chess
+    rank_to_index = {rank_: index_ for index_, rank_ in zip(range(8), "87654321")}  # translate rank in chess to range index
 
     def __init__(self):
         """Initialize a chessboard with a new game congifuration."""
 
-        self.matrix: list[list[Piece | None]] = [[None] * 8 for _ in range(8)]
+    #   Empty squares contain None in place of a Piece object.
+        self.board: list[list[Piece | None]] = [[None for _ in range(8)] for _ in range(8)]
```

Comment:
I don't think it is a good practice to name this attribute as `self.board` given the fact that the class is named `Board`. It complicates things.

Also why replace the elegant `[None] * 8` with a for loop with range? For loops with range are not nice in general.

Reply:
### Attribute name: `self.board`
In this case because basically the object and its one attribute are one and the same entity, ontologically.

Reply:
### Double list comprehension
In this case, I prefer notational uniformity to minimalism. List comprehensions are already a form of minimal notation, and the double list comprehension is just as clear and self-documenting as the inner shorthand `[None]*8` notation.

Reply:
@prototact could you share your thoughts on these two issues?